source/lib.civet

```diff
@@ -2538,6 +2561,50 @@ function processAssignments(statements): void {
     })
 }
 
+function processBlocks(statements): void
+  insertSemicolon(statements)
+  gatherRecursive statements, .type is "BlockStatement"
+  .forEach ({ expressions }) ->
+    processBlocks expressions
+
+/**
+* Avoid automatic continuation onto lines that start with
+* certain characters by adding an explicit semicolon. See
+* https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#automatic_semicolon_insertion
+*/
+function insertSemicolon(statements: StatementTuple[]): void
+  l := statements.length
+  statements.forEach (s, i) ->
+    if i < l - 1
+      if needsPrecedingSemicolon(statements[i + 1])
+        delim := s[2]
+        if !delim
+          s[2] = ";"
+        else if typeof delim is "string" and !delim.match(/;/)
+          s[2] = `;${delim}`
+
+function needsPrecedingSemicolon(exp)
+  let following: ASTNode
+  if Array.isArray exp
+    [,following] = exp
+  else
+    // Ideally this would be a StatementTuple but js only public/protected param items don't quite fit into that yet
+    // since the js: true is on the AST node the semicolon delimiters will stack up with empty nodes in between
+    following = exp
+
+  return false unless following
+
+  if Array.isArray following // for chained conditionals
+    return needsPrecedingSemicolon following[0]
+
+  switch following.type
+    when "ParenthesizedExpression", "ArrowFunction", "TemplateLiteral"
```

Comment:
I wonder about adding `IfExpression`, `DebuggerExpression`, `SwitchExpression`, `ThrowExpression`, `TryExpression` (a partial list from `makeLeftHandSideExpression`). Then again, if they've already been parenthesized at this stage, probably not needed.

Perhaps more importantly, I think `RegularExpressionLiteral` might need to be on this list; MDN lists this as an explicit case, and so our old code looked for leading `/`.